### ISSUE TITLE
Password Variable Not Working

### ISSUE THREAD

**Issue by Nitrocrusher** [mornedhels/enshrouded-server#66]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Used image version

Latest

### Used docker compose file or docker command

docker run -d --name enshrouded \
  --hostname enshrouded \
  --restart=unless-stopped \
  -p 15637:15637/udp \
  -v ./game:/opt/enshrouded \
  -e SERVER_NAME="My Server Name" \
  -e SERVER_PASSWORD="My Password" \
  -e UPDATE_CRON="*/30 * * * *" \
  -e PUID=4711 \
  -e PGID=4711 \
  mornedhels/enshrouded-server:latest

### What you expected to see, versus what you actually saw

I used the suggested commands, and the json shows the password and password variable, but does not ask for one when joining the server.

### Logs from the container

[mornedhels-enshrouded-server-1.csv](https://github.com/user-attachments/files/16199833/mornedhels-enshrouded-server-1.csv)

**Comment by Nitrocrusher** [comment]
After much searching you have to put a password for the roles as well for it to work.

**Comment by OnigamiSama** [comment]
Hey! I have the same error using the docker-compose example, can confirm that the server_password variable is not written in enshrouded_server.json

I made sure that all files are owned by my user (1000:1000) and the PUID and PGID are correct.

Following what said @Nitrocrusher , I tried to login using the userGroups password (wich is not the same as server_password) and it worked.

Is it supposed to work like this?
